client/src/hooks/authContext.js

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+
+const AuthContext = React.createContext();
+
+function AuthProvider({children}) {
+  const [user, setUser] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
```

Comment:
If feels like we're duplicating state here for the sake of easier updates/deconstruction and additional changes being required if we add user fields. Any time this state changes we would need to make this change in two locations. Since this state would only change on login, logout and the UserProfile update form, I propose that the update form has it's own local state initialized using the fields from `user` in `AuthProvider`. Then when the user hits save, we send a PUT request which can return the updated user object. Then take that updated user object from the response and update the `user` state in the `AuthProvider`. 

If destructuring is the concern we can just grab specific fields with `{user : {username, email, etc.} = useAuth()`

Reply:
So i knew of the alternate approach to put local state in UserProfile and can do it that way, but I assumed it doesn't hurt to put it all user state in Auth Provider. I'm not quite understanding why this way isn't ideal.
"Any time this state changes we would need to make this change in two locations." I am not quite understanding this. Can you elaborate?